juju/juju
# remove instanceCharmProfileData doc after subordinate charm lxd profile applied when unit created

Requires PR 9420 to land.

## Description of change

Remove the machines instanceCharmProfileData after the first time a subordinate charm's lxd profile is applied.  Normally removed during upgrade-charm otherwise.

## QA steps

1. juju bootstrap localhost
2. juju deploy ./testcharms/charm-repo/quantal/lxd-profile-alt  --to lxd -n 2
3. juju deploy ./testcharms/charm-repo/quantal/lxd-profile-subordinate
3. juju add-relation lxd-profile-subordinate lxd-profile-alt
4. juju debug-log -m controller --replay | grep "Removing instance charm profile data"
    lxd-profile-subordinate should be listed.
4. check the controller db, instanceCharmProfileData collection should be empty.
